### PR TITLE
Add docs on serving with Llama Stack

### PR DESCRIPTION
**Pull request by terrytangyuan**
See guide in https://github.com/meta-llama/llama-stack/blob/main/docs/source/getting_started/distributions/self_hosted_distro/remote_vllm.md

### CODE REVIEW

**Review by russellb — CHANGES_REQUESTED** [review]
want to mention the inline integration, as well?

https://github.com/meta-llama/llama-stack/tree/main/llama_stack/providers/inline/inference/vllm

**Comment by terrytangyuan** [comment]
Definitely! I'll update this in a bit. Let me see if I can find the relevant guide. I wasn't sure if that's available yet.

**Comment by terrytangyuan** [comment]
@russellb I couldn't find a guide yet but I have added a reference to the inline implementation.

File: docs/source/serving/serving_with_llamastack.rst [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+.. code-block:: yaml
+
+    inference:
+      - provider_id: vllm0
+        provider_type: remote::vllm
+        config:
+          url: http://127.0.0.1:8000
+
+Please refer to `this guide <https://github.com/meta-llama/llama-stack/blob/main/docs/source/getting_started/distributions/self_hosted_distro/remote_vllm.md>`_ for more details on this remote vLLM provider. An `inline vLLM provider <https://github.com/meta-llama/llama-stack/tree/main/llama_stack/providers/inline/inference/vllm>`_ is also available.
```
Comment (russellb) [inline docs/source/serving/serving_with_llamastack.rst:24]:
```suggestion
Inference using OpenAI Compatible API
-------------------------------------

Then start Llama Stack server pointing to your vLLM server with the following configuration:

.. code-block:: yaml

    inference:
      - provider_id: vllm0
        provider_type: remote::vllm
        config:
          url: http://127.0.0.1:8000

Please refer to `this guide <https://github.com/meta-llama/llama-stack/blob/main/docs/source/getting_started/distributions/self_hosted_distro/remote_vllm.md>`_ for more details on this remote vLLM provider.

Inference via Embedded vLLM
---------------------------

An `inline vLLM provider
<https://github.com/meta-llama/llama-stack/tree/main/llama_stack/providers/inline/inference/vllm>`_
is also available. This is a sample of configuration using that method:

.. code-block:: yaml

    inference
      - provider_type: vllm
        config:
          model: Llama3.1-8B-Instruct
          tensor_parallel_size: 4
```

**Review by russellb — CHANGES_REQUESTED** [review]
Suggested an addition of sample config

**Comment by terrytangyuan** [comment]
Thanks! Looking good.

**Review by russellb — APPROVED** [review]
Other than adding `Signed-off-by` to the commits so DCO will pass, this lgtm now. Thanks!

Feel free to add my `Signed-off-by` to the co-authored commit with my addition.

**Review by russellb — CHANGES_REQUESTED** [review]
Actually, I think you need one more change. Right now you'll get an error that this page is not listed anywhere:

```
checking consistency... /Users/rbryant/src/vllm/docs/source/serving/serving_with_llamastack.rst: WARNING: document isn't included in any toctree
```

I would add this to  `docs/source/serving/integrations.rst`

**Comment by terrytangyuan** [comment]
Should be fixed now!

**Comment by terrytangyuan** [comment]
Build failures seem unrelated.

**Review by russellb — APPROVED** [review]
thanks!

**Review by youkaichao — APPROVED** [review]
thanks for the integration!

**Pull request merged**